### ISSUE TITLE
win_find failed to check some files, these files were ignored and will not be part of the result output

### ISSUE THREAD

**Issue by aventrax** [ansible-collections/ansible.windows#101]
##### SUMMARY

Using win_find, even with become=true to become NT AUTHORITY\SYSTEM, the module fails to find any result. I think because of the dimension of the folder, because with smaller directories it seems to work, sometimes.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

```
win_find
```

##### ANSIBLE VERSION

```paste below
ansible 2.9.13
  config file = /home/matteo/.ansible.cfg
  configured module search path = ['/home/matteo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/python-virtual-envs/default/lib/python3.7/site-packages/ansible
  executable location = /opt/python-virtual-envs/default/bin/ansible
  python version = 3.7.3 (default, Dec 20 2019, 18:57:59) [GCC 8.3.0]
```

I just updated ansible. With 2.9.6 the issue was there.

##### OS / ENVIRONMENT

Target OS: Windows 2012R2 (latest updates).

##### STEPS TO REPRODUCE

```- name: Search for any .cfg file
  win_find:
    paths:
    - C:\TestFolder
    patterns:
    - '*.cfg'
    recurse: true
    get_checksum: false
  become: true
  become_user: SYSTEM
  become_method: runas
  register: o_find

- debug: var=o_find
```

##### EXPECTED RESULTS

Just having a list of ANY cfg file on any subfolder.

##### ACTUAL RESULTS

```ok: [10.1.1.51] => {
    "o_find": {
        "changed": false,
        "examined": 2,
        "failed": false,
        "files": [],
        "matched": 0,
        "warnings": [
            "win_find failed to check some files, these files were ignored and will not be part of the result output"
        ]
    }
}
```

There are 33 .cfg files on subfolders.

Folder size: 2.84 GB (3,055,742,976 bytes), contains 6,192 Files, 677 Folders --> NOT WORKING

```
ok: [10.1.1.51] => {
    "o_find": {
        "changed": false,
        "examined": 927,
        "failed": false,
        "files": [...],
        "matched": 28
    }
}
```

On a different folder, just a bit smaller, it works

Folder size: 261 MB (274,190,336 bytes), contains 778 Files, 149 Folders --> WORKING !!!

**Comment by jborean93** [comment]
The `win_find` folder has been rewritten for this collection and will be included in Ansible 2.10+. I would guess that this issue has been fixed as one of the original errors is that a failure on a single file brought the whole module down.

Could you try out the version in this collection, simply do the following to install the collection

```bash
ansible-galaxy collection install ansible.windows
```

Then as the module in your task set it to `ansible.windows.win_find`. While the collection is designed to work with Ansible 2.10+, the win_find module should be fine with 2.9+.

**Comment by aventrax** [comment]
In the meantime I have a playbook installing a list of application, looping on 10/12 elements, and on each loop the playbook performs a win_file searching for *.cfg files (2 minutes between each win_file). Sometimes it works, sometime it doesn't (same folder)!!

@jborean93 thanks! The new version is working like a charm!!!
